pulsar-io/alluxio/src/main/java/org/apache/pulsar/io/alluxio/sink/AlluxioStringSink.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.alluxio.sink;
+
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.io.core.KeyValue;
+import org.apache.pulsar.io.core.Sink;
+import org.apache.pulsar.io.core.annotations.Connector;
+import org.apache.pulsar.io.core.annotations.IOType;
+
+/**
+ * Alluxio sink that treats incoming messages on the input topic as Strings
+ * and write identical key/value pairs.
+ */
+@Connector(
+    name = "alluxio",
+    type = IOType.SINK,
+    help = "The AlluxioStringSink is used for moving records from Pulsar to Alluxio.",
+    configClass = AlluxioSinkConfig.class)
+public class AlluxioStringSink extends AlluxioAbstractSink<String, String> implements Sink<String> {
```

Comment:
Since Pulsar 2.8.0 you can use Sink<GenericObject> 
this way your Sink will work with every schema type.

you will have a `Record<GenericObject>` that will be able to deal transparently with:
- primitives (like Strings)
- schemaless (using byte[])
- structures AVRO/JSON
- KeyValue schema (with in turn both primitives and structures as keys and values)

you can see the docs here
https://pulsar.apache.org/docs/en/io-develop/#handling-schema-information

the problem with Sink<String> is that it will work only with Topics with String schema.
even if you want to support only String (and probably schemaless topics) I suggest you to use GenericObject in order to be ready for future enhancements

Reply:
@eolivelli I have addressed the above comments, please take a look again, 3q.